src/UI/Window.cpp

```diff
@@ -21,23 +21,19 @@
 
 
 
-Byte cWindow::m_WindowIDCounter = 0;
+char cWindow::m_WindowIDCounter = 1;
 
 
 
 
 
 cWindow::cWindow(WindowType a_WindowType, const AString & a_WindowTitle) :
-	m_WindowID(static_cast<char>((++m_WindowIDCounter) % 127)),
+	m_WindowID((++m_WindowIDCounter) % 127),
```

Comment:
This is a pre-increment so the ID of 0 won't be the first number assigned. Wrapping-around to 0 might be problematic though. Maybe this should be changed to 
```cpp
UInt8 cWindow::GetNextId()
{
    UInt8 Id = ++m_WindowIDCounter;
    return (Id == 0) ? ++m_WindowIDCounter : Id;
}

Reply:
I think `m_WindowID(((++m_WindowIDCounter) % 127) + 1),` should do the trick. The result of the modulo is in range [0, 126], add a 1 to get range [1, 127]. Whether to post- or pre-increment doesn't really matter here, just means the first `1` will be skipped.